pkg/gofr/container/sql_mock.go

```diff
@@ -107,17 +136,41 @@ func (m sqlMockDB) finish(t *testing.T) {
 // ExpectSelect is not a direct method for mocking the Select method of SQL in go-mock-sql.
 // Hence, it expects the user to already provide the populated data interface field,
 // which can then be used within the functions implemented by the user.
-func (m *mockSQL) ExpectSelect(_ context.Context, _ interface{}, query string, args ...interface{}) {
+func (m *mockSQL) ExpectSelect(_ context.Context, value any, query string, args ...interface{}) *queryWithArgs {
 	sliceQueryWithArgs := make([]queryWithArgs, 0)
 
 	if m.queryWithArgs == nil {
 		m.queryWithArgs = sliceQueryWithArgs
 	}
 
-	qr := queryWithArgs{query, args}
+	qr := queryWithArgs{queryText: query, arguments: args}
+
+	fieldType := reflect.TypeOf(value)
+	if fieldType.Kind() == reflect.Ptr {
+		qr.value = value
+	}
 
 	sliceQueryWithArgs = append(sliceQueryWithArgs, qr)
 	m.queryWithArgs = append(sliceQueryWithArgs, m.queryWithArgs...)
+
+	return &m.queryWithArgs[0]
+}
+
+func (q *queryWithArgs) ReturnsResponse(value any) {
+	fieldType := reflect.TypeOf(q.value)
+	if fieldType == nil {
+		return
+	}
+
+	fieldType = fieldType.Elem()
+
+	valueType := reflect.TypeOf(value)
+
+	if fieldType == valueType {
+		q.value = value
+	} else {
+		q.value = nil
+	}
```

Comment:
I tend to use this way


```suggestion
	q.value = nil
	if fieldType == valueType {
		q.value = value
	}
```